test/auth/oauth-with-idtoken-validation.tests.js

```diff
@@ -145,6 +145,27 @@ describe('OAUthWithIDTokenValidation', function() {
       });
       oauthWithValidation.create(PARAMS, DATA);
     });
+    it('Returns unvalidated response when header.alg === HS256 and there is no user secret', function(done) {
+      var oauth = {
+        create: function() {
+          return new Promise(res => res({ id_token: 'foobar' }));
+        }
+      };
+      sinon.stub(jwt, 'verify', function(idtoken, getKey, options, callback) {
+        getKey({ alg: 'HS256' }, function(err, key) {
+          expect(err.message).to.contain(
```

Comment:
I think besides skipping the validation is also important to test that the `console.warn` was called with the right message